src/API.jl

```diff
@@ -848,6 +850,55 @@ function activate(f::Function, new_project::AbstractString)
     end
 end
 
+########
+# Undo #
+########
+
+struct UndoSnapshot
+    date::DateTime
+    project::Types.Project
+    manifest::Types.Manifest
```

Comment:
Why not just store blobs with the entire content of the files here? That way if someone had some formatting or comments and are unhappy that Pkg has messed them up / delete them, they can get them back at least.

Reply:
I don't think people have formatting or comments in their Project file because that is removed the first thing that happens on every Pkg operations.

We don't deal with the blob anywhere else in Pkg so it quite a few special cases to handle it.
Right now I can just set the `.project` and `.manifest` fields to the ones at the state after the undo/redo and everything falls out (diff printing, writing the project etc).

Reply:
Ok, fair enough.